File: Algorithms/206 - Reverse Linked List.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None
 
class Solution(object):
    def reverseList(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        #variables
        temp = list()
         
        it = head
        while it:
            temp.append(it.val)
            it = it.next
         
        it = head
        index = len(temp) - 1
        while it:
            it.val = temp[index]
            it = it.next
            index -= 1
             
        return head

File: cpp/206 - Reverse Linked List.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        if (!head || !head->next)
            return head;
            
        // var
        ListNode* prev = NULL;
        ListNode* result = NULL;
       
        while (head) {
            prev = result;
            result = new ListNode(head->val);
            result->next = prev;
            head = head->next;
        }
        return result;
    }
};